src/jit/morph.cpp

```diff
@@ -493,20 +493,31 @@ GenTreePtr Compiler::fgMorphCast(GenTreePtr tree)
                 if (shiftAmount->IsIntegralConst())
                 {
                     const ssize_t shiftAmountValue = shiftAmount->AsIntCon()->IconValue();
-                    assert(shiftAmountValue >= 0);
 
-                    if (shiftAmountValue >= 32)
+                    if (shiftAmountValue >= 64)
```

Comment:
Isn't this value dependent on architecture? So for x86, >=32, but for x64 >= 64?

Reply:
Or, not by architecture, but type of value being shifted. From the CLI spec:
```
Stack Transition:
…, value, shiftAmount => …, result
Description:
The shl instruction shifts value (int32, int64 or native int) left by the number of bits specified by shiftAmount. shiftAmount is of type int32 or native int. The return value is unspecified if shiftAmount is greater than or equal to the width of value.
```

Reply:
This code is only reachable when we have a long shift.

Reply:
As far as I can tell the whole code is under `if ((srcType == TYP_LONG) && ((dstType == TYP_INT) || (dstType == TYP_UINT)))` so we're always dealing with a 64 bit shift.